Resources/js/router.js

```diff
@@ -294,14 +294,17 @@ class Router {
         });
         // Foo-bar!
         url = this.context_.base_url + url;
+
+        const isPortInHost = host.indexOf(':' + port) > -1 || this.getHost() && this.getHost().indexOf(':' + port) > -1;
```

Comment:
The code could be simplified (ans the performance improved) by storing the result of `host || this.getHost()` in a variable, and reusing it everywhere.
Same for `(isPortInHost || '' === port ? '' : ':' + port)`

This is not 100% related to this PR.

Reply:
I agree, this could be optimized and I would be happy to do it, but I'm not a fan of doing refactors without the maintainer's approval. I know how this may complicate merging up patches from the lower version to current and overall maintenance.